### PR TITLE
[WWST-1150] DTH for 4 endpoint Ecolink Siren

### PR DESCRIPTION
**Pull request by MMateusiakS**
@greens @MAblewiczS  @DCzarneckaS DTH ready for review

### CODE REVIEW

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy
Comment (greens) [inline devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy]: We shouldn't need to synthesize this event here. The `basicGet` should take care of that for us.

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy
Comment (greens) [inline devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy]: you could use `endPoints.each` here

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy
Comment (greens) [inline devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy:176]: this and `encap` appear to be the same function

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy
Comment (greens) [inline devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy]: why not just have the child devices send their endpoint number instead of parsing it out of the DNI?

**Review by greens — CHANGES_REQUESTED** [review]
I think you'll want to look at the zwave-siren DTH. It's important that we use the "Alarm" capability. It will essentially be redundant with "Switch", but it's important for how the device is presented and used. Speaking of which, using the generic-switch ui-metadata won't work, because that will cause the device to show up as a switch instead of an alarm/siren. Use the `generic-siren` metadata instead.

**Comment by MMateusiakS** [comment]
Hey @greens , I have made changes according to your comments. Thanks for advice about Alarm capability! Waiting for second review.

File: devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy
Comment (greens) [inline devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy]: you've already defined `val` as `"off"` here, so you can just check for `!= 0`

Comment (greens) [reply devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy]: you could use another ternary operator here like you do for `value1`

**Review by greens — APPROVED** [review]

**Pull request merged**
